src/wavesurfer.ts

```diff
@@ -433,9 +433,7 @@ class WaveSurfer extends Player<WaveSurferEvents> {
       if (staticDuration) {
         resolve(staticDuration)
       } else {
-        this.mediaSubscriptions.push(
-          this.onMediaEvent('loadedmetadata', () => resolve(this.getDuration()), { once: true }),
-        )
+        this.on('decode', (duration) => resolve(duration), { once: true })
```

Comment:
Isn’t this self-referential? This event is emitted at the end of this very method (loadAudio).

Reply:
Sorry for pushing it up without thorough testing, I'll try again these few days.

Reply:
No worries! We’ll probably have to split it into two checks: for an externally passed duration here, and for a decoded duration further down after the decoding.